Source/Core/DolphinQt/Settings/USBDeviceAddToWhitelistDialog.cpp

```diff
@@ -174,3 +180,41 @@ void USBDeviceAddToWhitelistDialog::OnDeviceSelection()
   device_vid_textbox->setText(*vid);
   device_pid_textbox->setText(*pid);
 }
+
+void USBDeviceAddToWhitelistDialog::OnVIDTextboxTextChanged()
+{
+  QString vid_text(device_vid_textbox->text());
+  QString pid_text(device_pid_textbox->text());
+  int pid_text_count = pid_text.count();
+  int vid_text_count = vid_text.count();
+
+  if (vid_text_count == 4 && pid_text_count == 4)
```

Comment:
This could be extracted into a helper function, e.g. bool IsVIDAndPIDValid() const.

Reply:
But a const?